hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotCache.java

```diff
@@ -58,14 +65,32 @@ public class SnapshotCache implements ReferenceCountedCallback, AutoCloseable {
       "SnapshotCacheCleanupService";
 
   private final OMMetrics omMetrics;
+  private final ReadWriteLock lock;
+  private final Lock readLock;
+  private final Lock writeLock;
+  private int lockCnt;
+  private final Condition lockReleased;
+  private final Condition dbClosed;
+  private final long lockTimeout;
+  private final Map<Long, Map<UUID, Long>> snapshotRefThreadIds;
+  private final AtomicBoolean closed;
 
   public SnapshotCache(CacheLoader<UUID, OmSnapshot> cacheLoader, int cacheSizeLimit, OMMetrics omMetrics,
-                       long cleanupInterval) {
+                       long cleanupInterval, long lockTimeout) {
     this.dbMap = new ConcurrentHashMap<>();
     this.cacheLoader = cacheLoader;
     this.cacheSizeLimit = cacheSizeLimit;
     this.omMetrics = omMetrics;
     this.pendingEvictionQueue = ConcurrentHashMap.newKeySet();
+    this.lock = new ReentrantReadWriteLock();
+    this.readLock = lock.readLock();
+    this.writeLock = lock.writeLock();
+    this.snapshotRefThreadIds = new ConcurrentHashMap<>();
+    this.lockTimeout = lockTimeout;
```

Comment:
Let's print an INFO level log that the snapshot cache lock timeout is set to X ms.